### PR TITLE
Extended `get_site_collection` to read the sites from the hazard curves when available

### PR DESCRIPTION
**Pull request by micheles**
This is another step towards https://github.com/gem/oq-engine/issues/3466

### CODE REVIEW

File: openquake/commonlib/readinput.py [hunk 1]
```diff
@@ -1124,13 +1122,12 @@ def get_pmap_from_nrml(oqparam, fname):
...
     mesh = geo.Mesh(numpy.array(lons), numpy.array(lats))
-    sitecol = get_site_collection(oqparam, mesh)
     num_levels = sum(len(v) for v in imtls.values())
-    array = numpy.zeros((len(sitecol), num_levels))
+    array = numpy.zeros((len(mesh), num_levels))
     imtls = DictArray(imtls)
     for imt_ in hcurves_by_imt:
         array[:, imtls.slicedic[imt_]] = hcurves_by_imt[imt_]
-    return sitecol, ProbabilityMap.from_array(array, sitecol.sids)
+    return mesh, ProbabilityMap.from_array(array, range(len(mesh)))
```
Comment (ptormene) [inline openquake/commonlib/readinput.py]: Please update the docstring accordingly

File: openquake/commonlib/readinput.py [hunk 2]
```diff
@@ -1093,11 +1092,10 @@ def get_pmap_from_csv(oqparam, fname):
         mesh, hcurves = get_mesh_csvdata(
             csvfile, list(oqparam.imtls), num_values,
             valid.decreasing_probabilities)
-    sitecol = get_site_collection(oqparam, mesh)
-    array = numpy.zeros((len(sitecol), sum(num_values)))
+    array = numpy.zeros((len(mesh), sum(num_values)))
     for imt_ in hcurves:
         array[:, oqparam.imtls.slicedic[imt_]] = hcurves[imt_]
-    return sitecol, ProbabilityMap.from_array(array, sitecol.sids)
+    return mesh, ProbabilityMap.from_array(array, range(len(mesh)))
```
Comment (ptormene) [inline openquake/commonlib/readinput.py]: Please update the docstring accordingly

File: openquake/commonlib/readinput.py [hunk 3]
```diff
@@ -1057,22 +1072,6 @@ def get_gmfs(oqparam):
     return eids, gmfs
```
Comment (ptormene) [inline openquake/commonlib/readinput.py]: Please fix the docstring also here (`sitecol` is not returned)

File: openquake/commonlib/readinput.py (hunk 2 above)
Comment (micheles) [reply openquake/commonlib/readinput.py]: Done

File: openquake/commonlib/readinput.py (hunk 1 above)
Comment (micheles) [reply openquake/commonlib/readinput.py]: Done

**Pull request merged**